lib/rust/ensogl/core/src/display/scene.rs

```diff
@@ -1207,31 +1210,58 @@ impl Deref for Scene {
 }
 
 impl Scene {
+    /// Perform early phase of scene update. This includes updating camera and the layout of all
+    /// display objects. No GPU buffers are updated yet, giving the opportunity to perform
+    /// additional updates that affect the layout of display objects after the main scene layout
+    /// has been performed.
+    ///
+    /// During this phase, the layout updates can be observed using `on_updated` FRP events on each
+    /// individual display object. Any further updates to the scene may require the `update`
+    /// method to be manually called on affected objects in order to affect rendering during
+    /// this frame.
     #[profile(Debug)]
-    // FIXME:
-    #[allow(unused_assignments)]
-    pub fn update(&self, time: animation::TimeInfo) -> UpdateStatus {
-        if let Some(context) = &*self.context.borrow() {
-            debug_span!("Updating.").in_scope(|| {
+    pub fn early_update(&self, time: animation::TimeInfo) -> UpdateStatus {
+        if self.context.borrow().is_some() {
+            debug_span!("Early update.").in_scope(|| {
                 let mut scene_was_dirty = false;
                 self.frp.frame_time_source.emit(time.since_animation_loop_started.unchecked_raw());
                 // Please note that `update_camera` is called first as it may trigger FRP events
                 // which may change display objects layout.
-                scene_was_dirty = self.update_camera(self) || scene_was_dirty;
+                scene_was_dirty |= self.update_camera(self);
                 self.display_object.update(self);
-                scene_was_dirty = self.layers.update() || scene_was_dirty;
-                scene_was_dirty = self.update_shape() || scene_was_dirty;
-                scene_was_dirty = self.update_symbols() || scene_was_dirty;
+                UpdateStatus { scene_was_dirty, pointer_position_changed: false }
+            })
+        } else {
+            default()
+        }
+    }
+
+    /// Perform late phase of scene update. At this point, all display object state is being
+    /// committed for rendering. This includes updating the layer stack, refreshing GPU buffers and
+    /// handling mouse events.
+    #[profile(Debug)]
+    pub fn late_update(
+        &self,
+        time: animation::TimeInfo,
+        early_status: UpdateStatus,
+    ) -> UpdateStatus {
+        if let Some(context) = &*self.context.borrow() {
+            debug_span!("Late update.").in_scope(|| {
+                let UpdateStatus { mut scene_was_dirty, mut pointer_position_changed } =
+                    early_status;
+                scene_was_dirty |= self.layers.update();
+                scene_was_dirty |= self.update_shape();
+                scene_was_dirty |= self.update_symbols();
                 self.handle_mouse_over_and_out_events();
```

Comment:
Make sure this works (!!!!!!)

Reply:
It does work as I expected it to: https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=c86d4ff2c083c9085cd910db4f43b3c5